tools/test.py

```diff
@@ -535,6 +539,7 @@ def RunCommand(self, command, env):
   def Run(self):
     try:
       result = self.RunCommand(self.GetCommand(), {
+        "TEST_SERIAL_ID": "%d" % self.serial_id,
```

Comment:
IMO this should be something that is traceable to the tested file.
maybe something like:
```py
        "TEST_SERIAL_ID": "%d" % re.sub('\W', '-', self.path),
```
This should also eliminate the need for changes in `ProgressIndicator`

Reply:
That's a good change, thanks! The number needs to stay for when using `--repeat`.

Reply:
Had to take this back because of http://docs.libuv.org/en/v1.x/pipe.html#c.uv_pipe_bind.

Would be good to have the test name in the tmpdir, but I don't see an easy way around this. With this PR, the tmpdir either gets removed correctly or the test that is using it will fail and print information. In any case, it's something that we didn't have before.

@refack let me know if you feel strongly about this and have some suggestion.